server/springboot/src/main/java/com/styra/tickethub_springboot/dao/model/LatencyMeasuringHTTPClient.java

```diff
@@ -27,17 +29,23 @@
  */
 public class LatencyMeasuringHTTPClient extends OPAHTTPClient {
 
+    Logger logger;
+
+    public LatencyMeasuringHTTPClient(Logger logger) {
```

Comment:
It's not normal to pass the logger into the constructor, define the logger in the class with the class name as the name of the logger

Reply:
But then how will the caller change what logger to use?

Reply:
Each class has it's own logger, no need to change it.  Loggers are controlled by central configuration.  A normal configuration might be something like `com.styra.opa=trace` (which would turn on trace logging for all loggers that start with that package), this is why we use the class name as the logger name.  You can change the log level for a single class or an entire package, gives you very fine control of logging.

Reply:
How would you change the log level at runtime then? That seems like it would make it quite difficult to test. On the opa-java PR, I rely on being able to change the logger for a particular class in order to test that the right log messages show up.

Reply:
Resolving this as discussion has moved to opa-java.